io/net/netperf_test.py

```diff
@@ -0,0 +1,130 @@
+        interfaces = netifaces.interfaces()
+        self.iface = self.params.get("Iface", default="")
+        self.peer_ip = self.params.get("PEERIP", default="")
+        if self.iface not in interfaces:
+            self.skip("%s interface is not available" % self.iface)
+        if self.peer_ip == "":
+            self.skip("%s peer machine is not available" % self.peer_ip)
+        self.timeout = self.params.get("timeout", default="600")
+        self.netperf_run = self.params.get("NETSERVER_RUN", default="0")
+        self.netperf = os.path.join(self.srcdir, 'netperf')
+        print self.netperf
+        tarball = self.fetch_asset('ftp://ftp.netperf.org/netperf/'
+                                   'netperf-2.7.0.tar.bz2', expire='7d')
+        archive.extract(tarball, self.netperf)
+        self.version = os.path.basename(tarball.split('.tar.')[0])
+        print self.version
+        self.neperf = os.path.join(self.netperf, self.version)
+        print self.neperf
+        tmp = "scp -r %s root@%s:" % (self.neperf, self.peer_ip)
+        if process.system(tmp, shell=True, ignore_status=True) != 0:
+            self.skip("unable to copy the netperf into peer machine")
+        tmp = "cd /root/%s;./configure ppc64le;make" % self.version
+        cmd = "ssh %s \"%s\"" % (self.peer_ip, tmp)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("test failed because command failed in peer machine")
+        time.sleep(5)
+        os.chdir(self.neperf)
+        process.system('./configure ppc64le', shell=True)
+        build.make(self.neperf)
+        self.perf = os.path.join(self.neperf, 'src', 'netperf')
+
+    def test(self):
+        """
+        netperf test
+        """
+        if self.netperf_run == 0:
+            tmp = "chmod 777 /root/%s/src" % self.version
+            cmd = "ssh %s \"%s\"" % (self.peer_ip, tmp)
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.fail("test failed because netserver not available")
+            cmd = "ssh %s \"/root/%s/src/netserver\"" % (self.peer_ip,
+                                                         self.version)
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.fail("test failed because netserver not available")
+            else:
+                self.netperf_run = 1
+        time.sleep(5)
+        msg = "timeout %s %s -H %s" % (
+            self.timeout, self.perf, self.peer_ip)
+        if process.system(msg, shell=True, ignore_status=True) != 0:
+            self.fail("test failed because netperf not working")
+        msg = "timeout %s %s -H %s -t UDP_STREAM -- -m 63000" % \
+              (self.timeout, self.perf, self.peer_ip)
+        if process.system(msg, shell=True, ignore_status=True) != 0:
+            self.fail("test failed because netperf not working")
+        msg = "timeout %s %s -H %s -t TCP_RR" % \
```

Comment:
```
info = "test failed because netperf not working"
for each in ["", "UDP_STREAM -- -m 63000", "TCP_RR", "UDP_RR"]:
    cmd_check(msg + "-t %s" % each, info)
```